src/renderers/dom/shared/DOMProperty.js

```diff
@@ -208,22 +219,51 @@ var DOMProperty = {
   getPossibleStandardName: __DEV__ ? {autofocus: 'autoFocus'} : null,
 
   /**
-   * All of the isCustomAttribute() functions that have been injected.
-   */
-  _isCustomAttributeFunctions: [],
-
-  /**
-   * Checks whether a property name is a custom attribute.
+   * Checks whether a property name is a writeable attribute.
    * @method
    */
-  isCustomAttribute: function(attributeName) {
-    for (var i = 0; i < DOMProperty._isCustomAttributeFunctions.length; i++) {
-      var isCustomAttributeFn = DOMProperty._isCustomAttributeFunctions[i];
-      if (isCustomAttributeFn(attributeName)) {
+  shouldSetAttribute: function(name, value) {
+    if (
+      DOMProperty.isReservedProp(name) ||
+      DOMProperty.aliases.hasOwnProperty(name)
+    ) {
+      return false;
+    }
+
+    if (DOMProperty.properties.hasOwnProperty(name)) {
+      return true;
+    }
+
+    if (value === null) {
+      return true;
+    }
+
+    if (name.toLowerCase() !== name) {
+      return false;
+    }
```

Comment:
We could also add a progressive warning system so that if you provide the same string with different casing, like two casings of the same attribute, we'd warn on that too.